### PR TITLE
Fix clojure-find-ns when there's a ns form inside a string

### PR DESCRIPTION
**Pull request by Malabarba**
**Replace this placeholder text with a summary of the changes in your PR.**

-----------------

Before submitting a PR mark the checkboxes for the items you've done (if you
think a checkbox does not apply, then leave it unchecked):

- [x] The commits are consistent with our [contribution guidelines][1].
- [x] You've added tests (if possible) to cover your change(s). Bugfix, indentation, and font-lock tests are extremely important!
- [ ] You've run `M-x checkdoc` and fixed any warnings in the code you've written.
- [x] You've updated the changelog (if adding/changing user-visible functionality).
- [ ] You've updated the readme (if adding/changing user-visible functionality).

Thanks!

[1]: https://github.com/clojure-emacs/clojure-mode/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: clojure-mode.el [hunk 1]
```diff
@@ -1894,19 +1907,16 @@ no namespaces above point, return the first one in the buffer.
...
-                  ;; Move to top-level to avoid searching from inside ns
-                  (ignore-errors (while t (up-list nil t t)))
-
-                  ;; The closest ns form above point.
-                  (when (or (re-search-backward clojure-namespace-name-regex nil t)
-                            ;; Or any form at all.
-                            (and (goto-char (point-min))
-                                 (re-search-forward clojure-namespace-name-regex nil t)))
-                    (match-string-no-properties 4))))))
+    (let* ((ns (save-excursion
```
Comment (bbatsov) [inline clojure-mode.el]: Seems to me a regular `let` would do here.

File: clojure-mode.el [hunk 2]
```diff
@@ -1886,6 +1886,19 @@ the cached value will be updated automatically."
 (defvar-local clojure-cached-ns nil
   "A buffer ns cache used to speed up ns-related operations.")
 
+(defun clojure--find-ns-in-direction (dir)
```
Comment (bbatsov) [inline clojure-mode.el]: Every time I see `dir` I think of directory, but that's not a big deal. It's a good idea to extract this.

File: clojure-mode.el (hunk 1 above)
Comment (Malabarba) [reply clojure-mode.el]: done

File: clojure-mode.el (hunk 2 above)
Comment (Malabarba) [reply clojure-mode.el]: No reason to be lazy anyway. Changed it to direction.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**